### ISSUE TITLE
False Positive: FailedMount events generated even when the pod comes up fine

### ISSUE THREAD

**Issue by tsunny** [kubernetes/kubernetes#99475]
#### What happened:
When we deploy our pods that use service-accounts. They randomly produce FailedMount events which looks something like this:

`MountVolume.SetUp failed for volume "default-token-vagiw" : failed to sync secret cache: timed out waiting for the condition
`
#### What you expected to happen:
No events when mounts work - so we can use it to alert for any failures.

#### How to reproduce it (as minimally and precisely as possible):
Creating a pod randomly gets failed event

#### Environment:
- Kubernetes version - GitVersion:"v1.19.8", GitCommit:"fd5d41537aee486160ad9b5356a9d82363273721"
- Cloud provider or hardware configuration: AWS
- OS (e.g: `cat /etc/os-release`): Ubuntu 18.04

Not sure how to debug this. The pods come up fine. It is really hard to reproduce. It could be great if the mounting could retry before sending the event.

**Comment by wzshiming** [comment]
Hi, do you know how to reproduce it?

**Comment by neolit123** [comment]
yes, please include reproduction steps with a minimal Pod YAML example.

/sig auth storage

**Comment by enj** [comment]
In case anyone goes looking for where the error is coming from:

https://github.com/kubernetes/kubernetes/blob/0f88bbe9b1fcb2138cd668948d75358c1a1d9858/pkg/kubelet/util/manager/watch_based_manager.go#L188

which funnels up to:

https://github.com/kubernetes/kubernetes/blob/6a7222cf4e708801624e77336b4e1e0a9bbfed67/pkg/volume/util/operationexecutor/operation_generator.go#L667-L691

**Comment by hermanbanken** [comment]
Is this related to #70044 maybe? Same piece of code at least.

**Comment by zoramthanga** [comment]
We are hitting this issue intermittently on both EKS and AKS with K8s 1.18. What's the best way to handle it? Keep waiting until the mounts eventually succeed?

The errors we're are something like

> failed to sync secret cache: timed out waiting for the condition

> failed to sync configmap cache: timed out waiting for the condition

**Comment by hermanbanken** [comment]
Still happening in our GKE clusters, on both 1.18 and 1.20.

**Comment by Dawgwe1gm** [comment]
Close all issues & accounts

**Comment by Dawgwe1gm** [comment]
Close all issues & accounts [Close all issues & accounts ]()
